Useheadings for selection

TobiasSchellhorn opened the pull request:
Pls add an option that i can use headings instead of namespace

samwilson committed:
General clean-up and tweaks:

  * Change to jQuery
  * Prevent form submission when no title has been entered
  * Code formatting
  * Allow more than one NEWPAGE form on a page
  * Declare as a 'block' type of output

samwilson committed:
Fix permissions of script.js file to permit execution.

samwilson committed:
Added languages from addnewpage wiki page, before deleting them from there.

samwilson committed:
Set option elements to just use 'selected', rather than 'selected=true'.

michael-dev committed:
add @PAGE@ and @NS@ for namespace template

samwilson committed:
Merge pull request #1 from michael-dev/master

Add @NS@ and @PAGE@ for namespace placeholders

michael-dev committed:
explain @PAGE@ and @NS@

samwilson committed:
Merge pull request #2 from michael-dev/master

Document @PAGE@ and @NS@

samwilson committed:
New release (and some code formatting).

samwilson committed:
Fix form-submission problem (issue #1).  Change from a POST form which has its action changed at submit-time, to a GET form which has a hidden id field which is changed at submit-time.  Also submit to the full DOKU_SCRIPT location, rather than just DOKU_BASE, because index.php doesn't forward query string.

samwilson committed:
Use jQuery's prop() rather than attr() to disable form elements.

samwilson committed:
Check for CREATE permission on target NS (issue #4).

Also cleaned up some variable names and locations, a bit of documentation, and simplified some slightly circuitous conditionals.

Release: 2013-05-17.

samwilson committed:
Parameter name not updated. I am a twit.

samwilson committed:
Encode the provided top-level NS with utf8_encodeFN() (as is done in a couple of dozen other places in DW). Thanks to Ryuu and tosmatt.

samwilson committed:
Updating version date.

samwilson committed:
Attempt to fix translation tool error:

> Hello,
>
> the import of your Add New Page Plugin failed.
> There was an error parsing a php language file.
>
> Filename: ar/settings.php
> Line: 13
> Message: Wrong key/value syntax, expected command end or eof

samwilson committed:
Code layout and minor fixes for most settings.php files. Removed Italiam and Russian which were just copies of the English one.

samwilson committed:
Added .gitattributes file, for normalizing line endings in PHP files (to CRLF).

samwilson committed:
Fix the .gitattributes file. Oops.

samwilson committed:
Coding standards fixes throughout codebase.

araname committed:
translation update

samwilson committed:
Merge pull request #8 from dokuwiki-translate/lang_update_217

Translation update (ko)

Commit:
translation update

samwilson committed:
Merge pull request #9 from dokuwiki-translate/lang_update_237

Translation update (nl)

brilanto committed:
translation update

samwilson committed:
Merge pull request #11 from dokuwiki-translate/lang_update_452

Translation update (eo)

bobzbobz committed:
translation update

samwilson committed:
Merge pull request #12 from dokuwiki-translate/lang_update_475

Translation update (da)

Klap-in committed:
minor changes, removed old stuff

Klap-in committed:
Code reformatting

Klap-in committed:
added support for newpagetemplate plugin

- inclusing some refactoring

Klap-in committed:
PHPDocs and trim namespace spaces.

Klap-in committed:
added name to select

Klap-in committed:
some escaping

alexgearbox committed:
translation update

samwilson committed:
Merge pull request #24 from dokuwiki-translate/lang_update_199

Translation update (ru)

mirko91 committed:
translation update

samwilson committed:
Merge pull request #30 from dokuwiki-translate/lang_update_23

Translation update (it)

samwilson committed:
Merge remote-tracking branch 'klapin/master'

samwilson committed:
Merge newpagetemplate feature, and fix up some coding standards items.

samwilson committed:
New release.

coastgnu committed:
translation update

samwilson committed:
Merge pull request #32 from dokuwiki-translate/lang_update_7

Translation update (de)

krmr committed:
translation update

Commit:
translation update

Commit:
translation update

samwilson committed:
Merge pull request #37 from dokuwiki-translate/lang_update_52

Translation update (ar)

samwilson committed:
Merge pull request #35 from dokuwiki-translate/lang_update_4

Translation update (de)

samwilson committed:
Merge pull request #36 from dokuwiki-translate/lang_update_14

Translation update (de-informal)

samwilson committed:
Include up to date original authorship and licence information, as per email from iDo.

samwilson committed:
Update release date.

sergey-art82 committed:
translation update

samwilson committed:
Merge pull request #42 from dokuwiki-translate/lang_update_25

Translation update (ru)

Commit:
Changed select menu to always include ancestors of any namespaces that show up in the menu,
even if the user does not have permissions for them.  In that case, they will be grayed out
but still be printed in the menu to avoid confusion and ambiguity.

Sitenl committed:
translation update

samwilson committed:
Merge pull request #46 from dokuwiki-translate/lang_update_13

Translation update (pt-br)

alexgearbox committed:
translation update

samwilson committed:
Merge pull request #47 from dokuwiki-translate/lang_update_28

Translation update (ru)

samwilson committed:
Merge pull request #43 from achern/master

Always print ancestor namespaces

samwilson committed:
Update metadata for new feature

Commit:
translation update

samwilson committed:
Merge pull request #48 from dokuwiki-translate/lang_update_47

Translation update (cs)

Commit:
translation update

samwilson committed:
Merge pull request #49 from dokuwiki-translate/lang_update_108

Translation update (pt)

nelidimitrova committed:
translation update

samwilson committed:
Merge pull request #52 from dokuwiki-translate/lang_update_157

Translation update (bg)

jeinwag committed:
fix ACL check when @NS@ is given as namespace

micgro42 committed:
Add bottom-margin: don't cling to next element

Fixes #56

micgro42 committed:
Clean whitespace from html source

Since all major browser now have developer tools with automatic source
code indentation adding further white-space to the source is
unnecessary and potentially a styling issue.

This retraces changes made in the DokuWiki core with the new release
Elenor of Tsort.

Co-authored-by: Damien Regad <dregad@mantisbt.org>

Conflicts:
    syntax.php

micgro42 committed:
Allow @PAGE@ to be used as part of the namespace

e.g. {{NEWPAGE>@PAGE@:feature}}

samwilson committed:
Merge pull request #54 from jeinwag/fix_acl_check

fix ACL check when @NS@ is given as namespace

samo079 committed:
translation update

samwilson committed:
Merge pull request #60 from dokuwiki-translate/lang_update_47

Translation update (fa)

hamstar commented:
I think you meant to pull request to the original repo by samwilson judging by your issue here: https://github.com/samwilson/dokuwiki-plugin-addnewpage/issues/7

micgro42 committed:
Replace the page anywhere in the string

Also clean the result.

samwilson committed:
Merge pull request #59 from cosmocode/pageAsNamespacePart

Allow @PAGE@ to be used as part of the namespace

arnehanssen committed:
translation update

ReanGD committed:
Add param "newpagevars" for plugin newpagetemplate

Fixes #66

algorys committed:
Fix PHP Warning "strpos(): Empty needle"

If $exclude is empty, it generates many PHP warnings:

```
 PHP Warning:  strpos(): Empty needle in /var/www/wiki/lib/plugins/addnewpage/syntax.php on line 249
```

This change fix it.

samwilson committed:
Merge pull request #67 from algorys/patch-1

Fix PHP Warning "strpos(): Empty needle"

samwilson committed:
Merge pull request #63 from dokuwiki-translate/lang_update_127

Translation update (no)

micgro42 committed:
fix: Always use id of main page

splitbrain committed:
more flexibility

* @INPUT@ placeholder can be used to reference the given input (useful
  to create a new namespace) makes #70 obsolete
* new ? syntax to overwrite config options from the syntax (defaults
  still come from the config setting)
* support for strftime placeholders in the namespace config. Allows to
  create a daily page for example
* New option autopage which hides the input field. Together with the new
  date placeholder this allows to create a daily page on a single button
  click

splitbrain committed:
updated version

splitbrain committed:
fix script to work on multiple forms again

splitbrain committed:
removed unneeded options assignment

Options are not used after the handler step, so no need to assign them
to the class member.

splitbrain committed:
use input event instead of keyup

This should also catch copy'n'paste actions.

https://caniuse.com/#feat=input-event

joergdd committed:
translation update

oleksiyfur committed:
translation update

alexbelchior committed:
translation update

rationality23 committed:
translation update

thienhau726 committed:
translation update

qezwan committed:
translation update

dregad committed:
Fix undefined array key error (#91)

dregad committed:
Bump version

Tagging a release with all the changes that were merged by @samwilson
since the last official one (2015-11-02).

dregad committed:
Update plugin info

dregad committed:
Merge commit 'c695c89c6220d13ad232a15a3eaf22575649997a' into pr72

dregad committed:
Update README.md

dregad committed:
Add .mailmap file

dregad committed:
Merge commit '5cc7d7358990f860049caa54fce63c5d6f7dd72e' into pr72

# Conflicts:
#	syntax.php

dregad committed:
Merge commit 'e5d3a18fa9f38378ed9f6b149ee11935d488b887' into pr72

dregad committed:
Avoid undefined array key error

dregad committed:
Add param "newpagevars" for plugin newpagetemplate

Fixes #66

UNTESTED adaptation of commit 9785b7d18e7da5a6454820073a551feb3c2a8617
to the new preg_match()-based parser.

dregad committed:
Merge commit '9418ebb801290cfb39b21fcf66b7ff91b61c4bba' into pr72

dregad committed:
Merge commit 'ed55cb4cce2354808a9d00f643c92c5c9c972933' into pr72

# Conflicts:
#	syntax.php

sizzname committed:
Fix caching (#79)

Users who can add pages have seen a cached page for users who are not 
allowed to add pages.

mw75 committed:
addnewpage with @NS@ in sidebar (#77)

If you use addnewpage with @NS@ in the sidebar, you get the namespace 
of the sidebar page (global ) and not the viewed one.

dregad committed:
Merge commit 'ccc8b5fd84846cd712c23144ff33056471b5091c' into pr72

# Conflicts:
#	syntax.php

dregad committed:
Move class PHPDoc to top of file

dregad committed:
Fix static analysis warnings

dregad committed:
Merge commit '7c0b9c8b0af497b353360b88a9fc849d39a54648' into pr72

# Conflicts:
#	syntax.php

dregad committed:
Replace deprecated resolve_pageid()

Use the PageResolver class instead, as documented [1].

Fixes #95

[1]: https://www.dokuwiki.org/devel:releases:refactor2021#refactoring_of_id_resolving

dregad committed:
Revert "Fix caching (#79)"

This reverts commit 52cbbe71e7149120aecbff6afe08360dba9a0142.

As @Klap-in pointed out [1] this change was incorrect as $disablecache
is passed by reference and updated by _htmlNamespaceInput().

Conflicts:
	syntax.php

[1]: https://github.com/dregad/dokuwiki-plugin-addnewpage/pull/79#discussion_r1187975731

dregad committed:
Merge commit '99f48d0389c2600fdcc50239a61bffd1aecc2861' into pr72

# Conflicts:
#	syntax.php

dregad committed:
Bump version

First release since @samwilson stopped maintaining the plugin in 2016.

It includes some housekeeping, translation updates as well as changes
from merging most of the pending pull requests.

dregad committed:
Fix incorrect merge resolution

In commit 22d9c079c34f0ef65334aeb535c0d954067f9d2a

dregad committed:
Fix static analysis warnings

dregad committed:
Fix static analysis warnings

dregad committed:
Replace legacy class by new SyntaxPlugin

dregad committed:
Whitespace

dregad committed:
Fix Array to string conversion warning

Regression introduced by incorrect implementation of newpagevars in
commit 0d90564c63508563e4f3c8776e96661181a11b99 when porting #66 - It
should be a string, not an array.

dregad committed:
Skip leading # in newpagevars parameter

The regex for newpagetemplate parameters parsing that was implemented in
commit 0d90564c63508563e4f3c8776e96661181a11b99 included the leading `#`
in the newpagevars parameter, preventing the newpagetemplate plugin from
processing the first parameter as `#@PARAM@` would not match the
expected `@PARAM@`.

Adapting the regex accordingly.

dregad committed:
French translation for autopage config

dregad committed:
PHPDoc

dregad committed:
Add placeholder text to page name input

Fixes #88

dokuwiki-translate committed:
translation update (hu) (#99)

Co-authored-by: Fekete Ádám Zsolt <fadam@egbcsoport.hu>

dokuwiki-translate committed:
translation update (ca) (#100)

Co-authored-by: Ingeborg <ingeborgpendley@gmail.com>

dregad committed:
Bump version to 2024-03-07

- Placeholder text (#88)
- Translations
- Code cleanup

dregad committed:
Revert "translation update (ca) (#100)"

The update actually introduced Russian spam.

This reverts commit 7400e97a0ee0f62c0526fccb4794f3f39a209628.

Fixes #101

dregad committed:
Bump version to 2024-09-14

Remove spam in Catalan translation (#101)

dregad committed:
Merge branch 'master' into pr72

# Conflicts:
#	plugin.info.txt
#	syntax.php

drmaxx committed:
translation update (de-informal)

Fixes #104, #106

dokuwiki-translate committed:
translation update (ru) (#103)

Co-authored-by: Aleksandr Selivanov <alexgearbox@yandex.ru>

dokuwiki-translate committed:
translation update (zh) (#107)

Co-authored-by: 小命Leaflet <2532846822@qq.com>

dregad committed:
Bump version to 2024-11-08

- Translation updates

dregad committed:
Fix PHP notice

dregad committed:
Parametric page name and other changes

Merge pull request #102 from dregad/pr72.

This is @splitbrain's original contribution (see PR #72), with merge
conflicts resolved and some additional code cleanup and fixes, including
French translation.

* a new `@INPUT@` placeholder can be used to reference the given input
  in namespace setup. For example you could use something like
  `{{NEWPAGE>foo:@INPUT@:start}}` to let users create new namespaces
  inside the `foo` namespace. This makes #70 obsolete
* a new `?` syntax allows to overwrite config options from the syntax
  (defaults still come from the config setting). Example:
  `{{NEWPAGE>foo:bar?nohide}}` overwrites the default `hide` setting.
* support for strftime placeholders in the namespace config. Allows to
  create a daily page for example. Example:
  `{{NEWPAGE>projekt:notes:%Y-%m-%d}}`
* New option autopage which hides the input field. Together with the new
  date placeholder this allows to create a daily page on a single button
  click. Example: `{{NEWPAGE>journal:%Y-%m-%d?noautopage}}`
* Includes changes of #71 (always use id of main page in replacements)

dregad committed:
Allow custom label for submit button

By specifying `{{NEWPAGE?label=xxxx}}`, it is now possible to set a
custom label on the Add New Page form's submit button, instead of using
the `Add page` default.

The specified value for the *label* parameter is a DokuWiki language
string, allowing localization; if no matching translation is found, then
`xxxx` will be displayed as-is.

Fixes #108

dregad committed:
Keep custom label's case if lang string not found

dregad committed:
Merge pull request #109 from dregad/custom-label

Custom button labels

dregad committed:
Translation update (fr)

- Use standard translation 'Catégorie' for 'namespace'
- Replace HTML entities by UTF-8 accented characters
- Minor fixes

dregad committed:
Bump version to 2024-11-16

- New @INPUT@ placeholder
- Allow overriding config options from the syntax
- Support for strftime placeholders in the namespace config
- New autopage option (use with strftime placeholders)
- Always use id of main page in replacements
- Allow customizing button labels
- Translation updates

dokuwiki-translate committed:
Translation update (pt-br) (#111)

Co-authored-by: Eduardo Mozart de Oliveira <eduardomozart182@gmail.com>

dregad committed:
Remove debugging code

dregad committed:
Fix page creation without @INPUT@ placeholder

When a plain `{{NEWPAGE}}` or {{NEWPAGE>ns?nohide}} is used, the form
did not create the new page as expected, because the javascript submit
event looked for an @INPUT@ placeholder that was not there, resulting in
the user's input not being added to the new page's name.

We now detect whether the placeholder is present, and if not append
the user's input to the namespace to generate the new page id.

Fixes #110

dregad committed:
Fix static analysis warnings

dregad committed:
Bump version to 2024-12-24

- Fix bug when creating pages without @INPUT@ placeholder
- Translation updates
- Code cleanup

dokuwiki-translate committed:
Translation update (de) (#114)

Co-authored-by: analogroboter <ropely@gmx.net>

dregad committed:
Bump version to 2025-03-07

- Translation updates

dokuwiki-translate committed:
translation update (cs) (#115)

Co-authored-by: Martin Růžička <martinr@post.cz>

dregad committed:
Bump version to 2025-04-22

- Translation updates

dokuwiki-translate committed:
translation update (ru) (#117)

Co-authored-by: Vyacheslav Strenadko <bryanskmap@ya.ru>

dregad committed:
Bump version to 2025-08-03

- Translation updates

dregad committed:
Allow replacement of multiple @INPUT@ placeholders

Fixes #113

dregad committed:
Only append page title if not empty

This fixes a regression introduced by fix for #110 (commit
2184b9df536ba26b3fce095eb7dbfd3499ee4eda), which prevented creation of
a new page when using the `autopage` option (the addnewpage form was
appending a `:` to the name, causing DokuWiki to create a "start" page
in a new namespace instead.

Fixes #118

dregad committed:
Bump version to 2025-08-13

- Allow replacement of multiple @INPUT@ placeholders
- Fix autopage creating a namespace instead of a page

dregad committed:
Allow disabling creation of subnamespaces

New 'createns' config option, enabled by default. If disabled, any `:`
in the page name will be replaced by `_` when determining the page name.

Fixes #120

dregad committed:
Coding guidelines

Apply DokuWiki's standard

dregad committed:
Bump version to 2025-09-06

- New `createns` option
- Code cleanup

dokuwiki-translate committed:
translation update (ru) (#122)

Co-authored-by: CL <chris.loewe@vodafone.de>

dregad committed:
Bump version to 2025-09-08

- Translation updates (de)

dokuwiki-translate committed:
translation update (fr)

PR #123

Co-authored-by: Schplurtz le Déboulonné <Schplurtz@laposte.net>

dregad committed:
Bump version to 2025-09-21

- Translation updates (fr)

dokuwiki-translate committed:
Translation update (ru) (#126)

Co-authored-by: Vyacheslav Strenadko <bryanskmap@ya.ru>

dregad committed:
Bump version to 2026-29-24

- Translation updates (ru)

dregad committed:
Bump version to 2026-04-24

- Translation updates (ru)

dokuwiki-translate committed:
Translation update (sk) (#127)

Co-authored-by: Wizzard <wizzardsk@gmail.com>

dregad committed:
Bump version to 2026-06-04

- Translation updates (sk)

xesarni committed:
Translation update (pl)

dregad committed:
Bump version to 2026-06-24

- Translation updates (pl)

tecnicoespaniol committed:
Translation update (es)

dregad committed:
Bump version to 2026-07-07

- Translation updates (es)
